code/Forms/UnsavedChangesIndicator.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace SilverStripe\Admin\Forms;
+
+use RuntimeException;
+use SilverStripe\Forms\FormAction;
+
+/**
+ * A visual indicator to show whether a form has unsaved changes.
+ *
+ * Note that this isn't strictly a FormAction as it doesn't perform any action when clicked.
+ * However, extending FormAction allows us to easily integrate with the existing form action
+ * architecture (e.g. placement in the form actions area).
+ */
+class UnsavedChangesIndicator extends FormAction
```

Comment:
```suggestion
class UnsavedChangesIndicator extends DatalessField
```
And add `use SilverStripe\Forms\DatalessField;`

The reason it doesn't work with `LiteralField` (which you noted in https://github.com/silverstripe/silverstripe-framework/pull/11905#discussion_r2540454285 doesn't work) is because `LiteralField` overrides the `Field()` and `FieldHolder()` methods to return the `$this->content` value. 

It doesn't make sense to have a settable content for this indicator, so we can just subclass `DatalessField` directly. I have tested this and it worked for me in cmsmain, assetadmin, siteconfig, and a modeladmin which is all I tried before concluding it works.

Note that just in case other code in these PRs currently relies on a `action_$action` name I added the below constructor. If any code does expect that name that code should be updated to not expect that name as this isn't an action (i.e. don't use this constructor - I'm just adding it so you know what I changed when I tested this).

```php
public function __construct($action)
{
    parent::__construct("action_$action", '');
}
```

Reply:
Updated